### ISSUE TITLE
It Crashed :(

### ISSUE THREAD

**Issue by 24thofjuly** [LyraelRayne/StacysWolves#26]
It worked earlier, and then I removed Doggy Talents. And now I keep crashing. I re-installed the mod, and deleted all of my config files to re-install those, but nothing has worked and I'm very confused. I haven't added a new mod or anything. 
Crash:
---- Minecraft Crash Report ----
// Don't be sad, have a hug! <3

Time: 01/06/16 17:02
Description: Initializing game

java.lang.NullPointerException: Initializing game
    at au.lyrael.stacywolves.registry.WolfRegistry.addSpawn(WolfRegistry.java:104)
    at au.lyrael.stacywolves.registry.WolfRegistry.registerForSpawning(WolfRegistry.java:86)
    at au.lyrael.stacywolves.CommonProxy.postInit(CommonProxy.java:47)
    at au.lyrael.stacywolves.StacyWolves.postInit(StacyWolves.java:91)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at cpw.mods.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:532)
    at sun.reflect.GeneratedMethodAccessor2.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
    at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
    at com.google.common.eventbus.EventBus.post(EventBus.java:275)
    at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:212)
    at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:190)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
    at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
    at com.google.common.eventbus.EventBus.post(EventBus.java:275)
    at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:119)
    at cpw.mods.fml.common.Loader.initializeMods(Loader.java:742)
    at cpw.mods.fml.client.FMLClientHandler.finishMinecraftLoading(FMLClientHandler.java:311)
    at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:552)
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:878)
    at net.minecraft.client.main.Main.main(SourceFile:148)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
## A detailed walkthrough of the error, its code path and all known details is as follows:

-- Head --
Stacktrace:
    at au.lyrael.stacywolves.registry.WolfRegistry.addSpawn(WolfRegistry.java:104)
    at au.lyrael.stacywolves.registry.WolfRegistry.registerForSpawning(WolfRegistry.java:86)
    at au.lyrael.stacywolves.CommonProxy.postInit(CommonProxy.java:47)
    at au.lyrael.stacywolves.StacyWolves.postInit(StacyWolves.java:91)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at cpw.mods.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:532)
    at sun.reflect.GeneratedMethodAccessor2.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
    at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
    at com.google.common.eventbus.EventBus.post(EventBus.java:275)
    at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:212)
    at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:190)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
    at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
    at com.google.common.eventbus.EventBus.post(EventBus.java:275)
    at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:119)
    at cpw.mods.fml.common.Loader.initializeMods(Loader.java:742)
    at cpw.mods.fml.client.FMLClientHandler.finishMinecraftLoading(FMLClientHandler.java:311)
    at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:552)

-- Initialization --
Details:
Stacktrace:
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:878)
    at net.minecraft.client.main.Main.main(SourceFile:148)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
    Minecraft Version: 1.7.10
    Operating System: Windows 7 (amd64) version 6.1
    Java Version: 1.8.0_25, Oracle Corporation
    Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
    Memory: 271212512 bytes (258 MB) / 836239360 bytes (797 MB) up to 1908932608 bytes (1820 MB)
    JVM Flags: 2 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx2G
    AABB Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
    IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
    FML: MCP v9.05 FML v7.10.99.99 Minecraft Forge 10.13.4.1558 Optifine OptiFine_1.7.10_HD_U_D4 62 mods loaded, 62 mods active
    States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
    UCHIJ   mcp{9.05} [Minecraft Coder Pack](minecraft.jar) 
    UCHIJ   FML{7.10.99.99} [Forge Mod Loader](forge-1.7.10-10.13.4.1558-1.7.10.jar) 
    UCHIJ   Forge{10.13.4.1558} [Minecraft Forge](forge-1.7.10-10.13.4.1558-1.7.10.jar) 
    UCHIJ   CodeChickenCore{1.0.7.46} [CodeChicken Core](minecraft.jar) 
    UCHIJ   Micdoodlecore{} [Micdoodle8 Core](minecraft.jar) 
    UCHIJ   NotEnoughItems{1.0.5.111} [Not Enough Items](NotEnoughItems-1.7.10-1.0.5.111-universal.jar) 
    UCHIJ   PotionExtensionCore{1.7.10-1.1.0} [PotionExtensionCore](minecraft.jar) 
    UCHIJ   lucky{5.1.0} [Lucky Block]([1-7-10]_Lucky_Block_v5-1-0.jar) 
    UCHIJ   DamageIndicatorsMod{3.2.0} [Damage Indicators]([1.7.10]DamageIndicatorsMod-3.2.0.jar) 
    UCHIJ   bspkrsCore{6.15} [bspkrsCore]([1.7.10]bspkrsCore-universal-6.15.jar) 
    UCHIJ   Treecapitator{1.7.10} [Treecapitator]([1.7.10]Treecapitator-universal-2.0.4.jar) 
    UCHIJ   ChickenShed{1.1.3} [Chicken Shed]([1.7.2+] ChickenShed %28v1.1.3%29.jar) 
    UCHIJ   More Crafting Recipes Mod{Minecraft 1.7.2} [More Crafting Recipes Mod]([2.2.0] More Crafting Recipes Mod.jar) 
    UCHIJ   adventurebackpack{1.7.10-0.8b} [Adventure Backpack](adventurebackpack-1.7.10-0.8c.jar) 
    UCHIJ   babyanimals{1.7.10-2} [Baby Animals Model Swapper](BabyAnimals-1.7.10-2.jar) 
    UCHIJ   BiblioCraft{1.11.4} [BiblioCraft](BiblioCraft[v1.11.4][MC1.7.10].jar) 
    UCHIJ   BiomesOPlenty{2.1.0} [Biomes O' Plenty](BiomesOPlenty-1.7.10-2.1.0.1462-universal.jar) 
    UCHIJ   candycraftmod{Beta 1.3} [CandyCraft](CandyCraft-1.3.jar) 
    UCHIJ   climatecontrol{0.6.beta57} [Climate Control](ClimateControl-0.6.beta58.jar) 
    UCHIJ   DynamicLights{1.3.9} [Dynamic Lights](DynamicLights-1.7.10.jar) 
    UCHIJ   DynamicLights_onFire{1.0.5} [Dynamic Lights Burning Entity Module](DynamicLights-1.7.10.jar) 
    UCHIJ   DynamicLights_creepers{1.0.4} [Dynamic Lights Creeper Module](DynamicLights-1.7.10.jar) 
    UCHIJ   DynamicLights_dropItems{1.0.8} [Dynamic Lights EntityItem Module](DynamicLights-1.7.10.jar) 
    UCHIJ   DynamicLights_entityClasses{1.0.1} [Dynamic Lights Entity Light Module](DynamicLights-1.7.10.jar) 
    UCHIJ   DynamicLights_mobEquipment{1.0.8} [Dynamic Lights Mob Equipment Light Module](DynamicLights-1.7.10.jar) 
    UCHIJ   DynamicLights_flameArrows{1.0.0} [Dynamic Lights Fiery Arrows Light Module](DynamicLights-1.7.10.jar) 
    UCHIJ   DynamicLights_floodLights{1.0.2} [Dynamic Lights Flood Light](DynamicLights-1.7.10.jar) 
    UCHIJ   DynamicLights_otherPlayers{1.0.8} [Dynamic Lights OtherPlayers Light Module](DynamicLights-1.7.10.jar) 
    UCHIJ   DynamicLights_thePlayer{1.1.4} [Dynamic Lights Player Light Module](DynamicLights-1.7.10.jar) 
    UCHIJ   eplus{3.0.2-d} [Enchanting Plus](EnchantingPlus-1.7.10-3.0.2-d.jar) 
    UCHIJ   etfuturum{1.5.5} [Et Futurum](Et Futurum-1.5.5.jar) 
    UCHIJ   golems{1.20} [Extra Golems](Extra-Golems[1.7.10]-1.20.jar) 
    UCHIJ   ExtrabiomesXL{3.16.4} [ExtrabiomesXL](extrabiomesxl_1.7.10-3.16.4.jar) 
    UCHIJ   fairylights{1.4.0} [Fairy Lights](fairylights-1.4.0-1.7.10.jar) 
    UCHIJ   fairymod{0.5} [Fairy Mod](fairymod-0.5-1.7.10.jar) 
    UCHIJ   FastCraft{1.21} [FastCraft](fastcraft-1.21.jar) 
    UCHIJ   grim3212core{V0.1 - 1.7.10} [Grim3212 Core](Grim3212 Core-V0.1 - 1.7.10.jar) 
    UCHIJ   fireplaces{V0.1 - 1.7.10} [Fireplaces](Fireplaces-V0.1 - 1.7.10.jar) 
    UCHIJ   F2L{1.7.10b} [Flesh2Leather](Flesh2Lether [1.7.10b].jar) 
    UCHIJ   IC2{2.2.727-experimental} [IndustrialCraft 2](industrialcraft-2-2.2.727-experimental.jar) 
    UCHIJ   GalacticraftCore{3.0.12} [Galacticraft Core](GalacticraftCore-1.7-3.0.12.350.jar) 
    UCHIJ   GardenContainers{1.7.10-1.7.0} [Garden Containers](GardenStuff-1.7.10-1.7.0.jar) 
    UCHIJ   GardenCore{1.7.10-1.7.0} [Garden Core](GardenStuff-1.7.10-1.7.0.jar) 
    UCHIJ   GardenStuff{1.7.10-1.7.0} [Garden Stuff](GardenStuff-1.7.10-1.7.0.jar) 
    UCHIJ   GardenTrees{1.7.10-1.7.0} [Garden Trees](GardenStuff-1.7.10-1.7.0.jar) 
    UCHIJ   GollumCoreLib{2.0.0} [Gollum Core Lib](Gollum-Core-Lib-1.7.10.jar) 
    UCHIJ   InventoryPets{1.4.3} [Inventory Pets](inventorypets-1.7.10-1.4.3-universal.jar) 
    UCHIJ   inventorytweaks{1.58-147-645ca10} [Inventory Tweaks](InventoryTweaks-1.58-147 %281%29.jar) 
    UCHIJ   IronChest{6.0.60.741} [Iron Chest](ironchest-1.7.10-6.0.60.741-universal.jar) 
    UCHIJ   JammyFurniture{5.0.0 [Build Smeagol]} [Jammy Furniture Mod](Jammy-Furniture-Reborn-Mod-1.7.10.jar) 
    UCHIJ   journeymap{5.1.4p1} [JourneyMap](journeymap-1.7.10-5.1.4p1-unlimited.jar) 
    UCHIJ   mermaidtail{0.2} [Mermaid Tail](mermaidtail-0.2-1.7.10.jar) 
    UCHIJ   movillages{1.2.1} [Mo' Villages](MoVillages-1.3.1.jar) 
    UCHIJ   cfm{3.4.8} [§9MrCrayfish's Furniture Mod](MrCrayfishFurnitureModv3.4.8%281.7.10%29.jar) 
    UCHIJ   Oceancraft{1.4.1} [Oceancraft](Oceancraft-1.4.1.jar) 
    UCHIJ   PetBat{1.3.8} [Pet Bat](PetBat-1.7.10.jar) 
    UCHIJ   pokeball{V0.1 - 1.7.10} [Pokeball](Pokeball-V0.1 - 1.7.10.jar) 
    UCHIJ   primitivemobs{1.0} [Primitive Mobs](primitivemobs-1.0c-1.7.10.jar) 
    UCHIE   stacywolves{1.0.7-1.7.10} [Stacy's Wolves Mod](Stacy'sWolvesMod-1.7.10-1.0.7.jar) 
    UCHIJ   ultimate_unicorn_mod{1.4.12} [Wings, Horns, and Hooves, the Ultimate Unicorn Mod!](ultimate_unicorn_mod-1.7.10-1.4.12.jar) 
    UCHIJ   Waila{1.5.10} [Waila](Waila-1.5.10_1.7.10.jar) 
    UCHIJ   witchery{0.24.1} [Witchery](witchery-1.7.10-0.24.1.jar) 
    GL info: ' Vendor: 'ATI Technologies Inc.' Version: '4.4.12874 Compatibility Profile Context 14.100.0.0' Renderer: 'ATI Radeon HD 5400 Series'
    Launched Version: 1.7.10
    LWJGL: 2.9.1
    OpenGL: ATI Radeon HD 5400 Series GL version 4.4.12874 Compatibility Profile Context 14.100.0.0, ATI Technologies Inc.
    GL Caps: Using GL 1.3 multitexturing.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Anisotropic filtering is supported and maximum anisotropy is 16.
Shaders are available because OpenGL 2.1 is supported.

```
Is Modded: Definitely; Client brand changed to 'fml,forge'
Type: Client (map_client.txt)
Resource Packs: [Black Wolf, Border Collie, Dalmatian, German Shepherd, Glen of Imaal Terrier, Golden Retriever, Gray Wolf, Light Gray Wolf, Nyan Cat, Ocelot, Pug, Red Fox, Tosa Inu, Weimaraner, White Wolf, AliPack.zip]
Current Language: English (US)
Profiler Position: N/A (disabled)
Vec3 Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
Anisotropic Filtering: Off (1)
OptiFine Version: OptiFine_1.7.10_HD_U_D4
Render Distance Chunks: 7
Mipmaps: 4
Anisotropic Filtering: 1
Antialiasing: 0
Multitexture: false
OpenGlVersion: 4.4.12874 Compatibility Profile Context 14.100.0.0
OpenGlRenderer: ATI Radeon HD 5400 Series
OpenGlVendor: ATI Technologies Inc.
CpuCount: 3
```

**Comment by LyraelRayne** [comment]
Looks like perhaps sometimes biome dictionary entries can have an empty biome.

I have added a check for this and caused the mod not to try and register wolves to spawn in these biomes.

If you could try the attached mod JAR (make sure you unzip it first and delete/rename the old one)  and let me know. A copy of the fml-client-latest.log (or fml-server-latest.log if running a server) with the new JAR that would be helpful as I added some extra logging to tell me exactly what triggered this crash.
[Stacy'sWolvesMod-1.7.10-1.0.7b.jar.zip](https://github.com/LyraelRayne/StacysWolves/files/294559/Stacy.sWolvesMod-1.7.10-1.0.7b.jar.zip)

**Comment by 24thofjuly** [comment]
I think it was the Mo Villages mod, because I removed all the mods and added them one by one, loading the game as I go, and it seemed to crash when I added the Mo Villages mod.

**Comment by LyraelRayne** [comment]
Did you try the 1.0.7b version of Stacy's Wolves? It's good to know which mod is causing it, but I hope to fix the mod on my end to prevent such crashes if possible.

**Comment by LyraelRayne** [comment]
Closing as we haven't heard any response in a while. This issue is hopefully fixed in version 1.0.8.
